pkg/util/validation/limits.go

```diff
@@ -667,6 +676,26 @@ func (o *Overrides) AlertmanagerMaxAlertsSizeBytes(userID string) int {
 	return o.GetOverridesForUser(userID).AlertmanagerMaxAlertsSizeBytes
 }
 
+func (o *Overrides) DisabledRuleGroups(userID string) DisabledRuleGroups {
+	if o.tenantLimits != nil {
+		l := o.tenantLimits.ByUserID(userID)
+		if l != nil {
+			disabledRuleGroupsForUser := make(DisabledRuleGroups, len(l.DisabledRuleGroups))
+
+			for i, disabledRuleGroup := range l.DisabledRuleGroups {
+				disabledRuleGroupForUser := DisabledRuleGroup{
+					Namespace: disabledRuleGroup.Namespace,
+					Name:      disabledRuleGroup.Name,
+					User:      userID,
+				}
+				disabledRuleGroupsForUser[i] = disabledRuleGroupForUser
+			}
+			return disabledRuleGroupsForUser
+		}
+	}
+	return DisabledRuleGroups{}
```

Comment:
Should we return the default limits here? I think someone may wanna set this on the config (instead runtimeconfig) as well.

Reply:
I went back and forth on this one. Since `user` would be required to disable the rule groups to avoid accidentally disabling rule groups from other users that happen to have the same name and namespace, I thought it might be better to allow this config only from runtime config. If we want to allow config, then we need to have `user` field exposed [here](https://github.com/cortexproject/cortex/pull/5521/files#diff-1e0a07c0c1ea87ab5a582d79c587a73a9b8e0680d89c61d5c94be106823a4a4eR44) so it can be set. If we did that, then someone can configure `user` under runtime config which is already a map of [user:overrides] which can be confusing

Reply:
Ok! +1